examples/server/server.cpp

```diff
@@ -603,14 +608,451 @@ void get_req_parameters(const Request & req, whisper_params & params)
+                        task.params.n_threads, task.params.n_processors,
+                        task.params.language.c_str(),
+                        task.params.translate ? "translate" : "transcribe",
+                        task.params.tinydiarize ? "tdrz = 1, " : "",
+                        task.params.no_timestamps ? 0 : 1);
+                fprintf(stderr, "\n");
+            }
+
+            // run the inference
+            {
+                printf("Running whisper.cpp inference on %s\n", task.filename.c_str());
+                whisper_full_params wparams = whisper_full_default_params(WHISPER_SAMPLING_GREEDY);
+
+                wparams.strategy = task.params.beam_size > 1 ? WHISPER_SAMPLING_BEAM_SEARCH : WHISPER_SAMPLING_GREEDY;
+
+                wparams.print_realtime   = false;
+                wparams.print_progress   = task.params.print_progress;
+                wparams.print_timestamps = !task.params.no_timestamps;
+                wparams.print_special    = task.params.print_special;
+                wparams.translate        = task.params.translate;
+                wparams.language         = task.params.language.c_str();
+                wparams.detect_language  = task.params.detect_language;
+                wparams.n_threads        = task.params.n_threads;
+                wparams.n_max_text_ctx   = task.params.max_context >= 0 ? task.params.max_context : wparams.n_max_text_ctx;
+                wparams.offset_ms        = task.params.offset_t_ms;
+                wparams.duration_ms      = task.params.duration_ms;
+
+                wparams.thold_pt         = task.params.word_thold;
+                wparams.max_len          = task.params.max_len == 0 ? 60 : task.params.max_len;
+                wparams.split_on_word    = task.params.split_on_word;
+                wparams.audio_ctx        = task.params.audio_ctx;
+
+                wparams.debug_mode       = task.params.debug_mode;
+                wparams.tdrz_enable      = task.params.tinydiarize;
+                wparams.initial_prompt   = task.params.prompt.c_str();
+
+                wparams.greedy.best_of        = task.params.best_of;
+                wparams.beam_search.beam_size = task.params.beam_size;
+
+                wparams.temperature      = task.params.temperature;
+                wparams.no_speech_thold = task.params.no_speech_thold;
+                wparams.temperature_inc  = task.params.temperature_inc;
+                wparams.entropy_thold    = task.params.entropy_thold;
+                wparams.logprob_thold    = task.params.logprob_thold;
+
+                wparams.no_timestamps    = task.params.no_timestamps;
+                wparams.token_timestamps = !task.params.no_timestamps && task.params.response_format == vjson_format;
+                wparams.no_context       = task.params.no_context;
+                wparams.suppress_nst     = task.params.suppress_nst;
+
```

Comment:
Perhaps also add VAD parameters here similar to what is done in https://github.com/ggml-org/whisper.cpp/blob/2a4d6db7d90899aff3d58d70996916968e4e0d27/examples/server/server.cpp#L924-L932